File: StriverDSAsheet/linked_list/convertArr2LL.java
class Node {
    int data;
    Node next;

    Node(int data1,Node next1){
        this.data = data1;
        this.next = next1;
    }
    
    //if our next element is nullpointer
    Node(int data1){
        this.data = data1;
        this.next = null;
    }
};

private static Node convertArr2LL(int[] arr){
    Node head = new Node(arr[0]);
    Node mover = head;
    for(int i=1; i<arr.length; i++){
        Node tmp = new Node(arr[i]);
        mover.next = tmp;
        mover = mover.next; // mover = tmp;
    }
    return head;
}

public class LinkedList {
    public static void main(String[] args) {
        int[] arr = {12, 5, 8, 6};
        Node head = convertArr2LL(arr);
        System.out.print(head.data);
    }
}

File: StriverDSAsheet/linked_list/convert_arr2LL.cpp
#include<bits/stdc++.h>
using namespace std;

class Node {
    public:
    int data;
    Node* next;
    public:
    Node(int data1,Node* next1){
        data = data1;
        next = next1;
    }
    
    //if our next element is nullpointer
    public:
    Node(int data1){
        data = data1;
        next = nullptr;
    }
};

Node* convertArr2LL(vector<int> &arr){
    Node* head = new Node(arr[0]);
    Node* mover = head;
    for(int i=1; i<arr.size(); i++){
        Node* tmp = new Node(arr[i]);
        mover->next = tmp;
        mover = mover->next; // mover = tmp;
    }
    return head;
}

int main()
{
    vector<int> arr = {12, 5, 8, 7};
    Node* head = convertArr2LL(arr);
    cout << head->data;
    return 0;
}